core/domain/html_cleaner.py

```diff
@@ -192,3 +193,401 @@ def get_rte_components(html_string: str) -> List[ComponentsDict]:
+            empty.
+        ValidationError. SkillReview does not have text-with-value attribute.
+        ValidationError. SkillReview text-with-value attribute should not be
+            empty.
+        ValidationError. SkillReview does not have skill_id-with-value
+            attribute.
+        ValidationError. SkillReview skill_id-with-value attribute should not be
+            empty.
+        ValidationError. Video does not have start-with-value attribute.
+        ValidationError. Video start-with-value attribute should not be empty.
+        ValidationError. Video does not have end-with-value attribute.
+        ValidationError. Video end-with-value attribute should not be empty.
+        ValidationError. Start value is greater than end value.
+        ValidationError. Video does not have autoplay-with-value attribute.
+        ValidationError. Video autoplay-with-value attribute should be boolean.
+        ValidationError. Video does not have video_id-with-value attribute.
+        ValidationError. Link does not have text-with-value attribute.
+        ValidationError. Link does not have url-with-value attribute.
+        ValidationError. Link url-with-value attribute should not be empty.
+        ValidationError. Math does not have math_content-with-value attribute.
+        ValidationError. Math math_content-with-value attribute should not be
+            empty.
+        ValidationError. Math does not have raw_latex-with-value attribute.
+        ValidationError. Math raw_latex-with-value attribute should not be
+            empty.
+        ValidationError. Math does not have svg_filename-with-value attribute.
+        ValidationError. Math svg_filename-with-value attribute should not be
+            empty.
+        ValidationError. Math svg_filename attribute does not have svg
+            extension.
+        ValidationError. Tabs tag present inside another tabs or collapsible.
+        ValidationError. Collapsible tag present inside tabs or another
+            collapsible.
+    """
+    soup, empty_values = _initialize_bs4_and_empty_values_variable(html_data)
+    for tag in soup.find_all('oppia-noninteractive-image'):
+        if not tag.has_attr('alt-with-value'):
+            raise utils.ValidationError(
+                'Image tag does not have \'alt-with-value\' attribute.'
+            )
+
+        alt_value = utils.unescape_html(
+            tag['alt-with-value'])[1:-1].replace('\\"', '')
+        if len(alt_value.strip()) < 5:
+            raise utils.ValidationError(
+                'The length of the image tag \'alt-with-value\' '
+                'attribute value should be at least 5 characters.'
+            )
+
+        if not tag.has_attr('caption-with-value'):
+            raise utils.ValidationError(
+                'Image tag does not have \'caption-with-value\' attribute.'
+            )
+
+        caption_value = utils.unescape_html(
+                tag['caption-with-value'])[1:-1].replace('\\"', '')
```

Comment:
```suggestion
        caption_value = utils.unescape_html(
            tag['caption-with-value'])[1:-1].replace('\\"', '')
```

Reply:
Done

Reply:
Not done.

Reply:
Done, I missed that accidentally.
Thanks